### PR TITLE
MON-3355: build: build and publish test harness images in release workflow

### LINKED ISSUE #298: Automate building and publishing osd-e2e-test-harness image through CI

### PR DESCRIPTION
**Pull request by jan--f**
Fixes #298

### CODE REVIEW

**Comment by sthaha** [comment]
@jan--f , since this modifies release.yaml, could you also please provide link to the CI run on your local fork where this is tested ?

**Comment by jan--f** [comment]
@sthaha good point, run is at https://github.com/jan--f/observability-operator/actions/runs/6034088227

File: .github/osd-test-harness-publish/action.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - name: Registry Login
+    uses: docker/login-action@v2
+    with:
+      registry: quay.io
+      username: ${{ inputs.quay_login }}
+      password: ${{ inputs.quay_token }}
+
+  - uses: actions/checkout@v3
+    with:
+      fetch-depth: 0
```
Comment (danielmellado) [inline .github/osd-test-harness-publish/action.yaml:35]: Would we really need all history for all branches and tags?

File: .github/osd-test-harness-publish/action.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  go-version:
+    description: "go version"
+    required: true
+runs:
+  using: composite
+  steps:
+  - name: Setup Go environment
+    uses: actions/setup-go@v3
+    with:
+      go-version: ${{ inputs.go-version }}
```
Comment (sthaha) [inline .github/osd-test-harness-publish/action.yaml:19]: On an unrelated note, setup-go now allows pointing to `go.mod` - https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

**Comment by jan--f** [comment]
Waiting on a tide config change: https://github.com/openshift/release/pull/42910

File: .github/osd-test-harness-publish/action.yaml (hunk 2 above)
Comment (jan--f) [reply .github/osd-test-harness-publish/action.yaml:19]: oh very nice, I'll prep a PR

File: .github/osd-test-harness-publish/action.yaml (hunk 1 above)
Comment (jan--f) [reply .github/osd-test-harness-publish/action.yaml:35]: probably not for this action no. I'll check them all and push a PR. There are others that look suspect too.

**Pull request merged**